src/server/src/controllers/comment.controllers.ts

```diff
@@ -0,0 +1,108 @@
+import { Request, Response } from 'express';
+import Author from '../models/Author';
+import Comment from '../models/Comment';
+import Post from '../models/Post';
+import { AuthenticatedRequest } from '../types/auth';
+import { PaginationRequest } from '../types/pagination';
+
+const createComment = async (req: AuthenticatedRequest, res: Response) => {
+  if (req.params.comment_id) {
+    const comment_exists = await Comment.findOne({
+      where: { id: req.params.post_id },
+    });
+    if (comment_exists !== null) {
+      res.status(400).send({ error: 'Comment already exists' });
+      return;
+    }
+  }
+
+  const { comment, contentType } = req.body;
+  const commentStr = comment;
+
+  const post = await Post.findOne({
+    where: {
+      id: req.params.post_id,
+    },
+  });
+  const author = await Author.findOne({
+    where: {
+      id: req.params.id,
+    }
+  })
+  if (post === null) {
+    res.status(404).send();
+    return;
+  }
+  try {
+    const comment = await Comment.create({
+      ...(req.params.comment_id && { id: req.params.comment_id }),
```

Comment:
Again, not necessary. Look at what is required for the routes endpoint and the requirements doc

Reply:
Yep, same reason as the previous one. Removed!